### PR TITLE
Warn user about problematic locale settings

### PR DESCRIPTION
**Pull request by milch**
Some tools might silently fail or hang otherwise. See #10988

### CODE REVIEW

File: fastlane/lib/fastlane/cli_tools_distributor.rb [hunk 1]
```diff
@@ -22,6 +22,15 @@ def take_off
           print_bundle_exec_warning(is_slow: (Time.now - before_import_time > 3))
         end
 
+        unless ENV['LANG'].end_with?("UTF-8") || ENV['LC_ALL'].end_with?("UTF-8")
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/cli_tools_distributor.rb]: Does this compare need to be case insensitive? 🤔

Comment (milch) [reply fastlane/lib/fastlane/cli_tools_distributor.rb]: I'm not sure but I think it is better to err on the side of caution here: theoretically your locale has to match one of the outputs of `locale -a`, which use the uppercased `UTF-8` spelling

**Review by joshdholtz — APPROVED** [review]

**Comment by janpio** [comment]
Just noticed: The `bundle exec fastlane test` phase of the Circle build is full of this output now:

```
^D^D[07:32:19]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables
[07:32:19]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables
[07:32:19]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables
[07:32:19]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables
```
(and all through the test)

Intentional?

**Comment by milch** [comment]
Seems like we need to reconfigure circle? 😅

**Comment by milch** [comment]
@janpio See #11003, it's fixed now

**Comment by funnel20** [comment]
After updating from Fastlane 2.64.1 to 2.68.0 I get several WARNINGs

**Before**
Martijns-MacBook-Pro:Frameit Martijn$ fastlane frameit silver
[23:23:14]: Seems like launching fastlane takes a while - please run
[23:23:14]: 
[23:23:14]: $ [sudo] gem cleanup
[23:23:14]: 
[23:23:14]: to uninstall outdated gems and make fastlane launch faster
[23:23:14]: Alternatively it's recommended to start using a Gemfile to lock your dependencies
[23:23:14]: To get started with a Gemfile, run
[23:23:14]: 
[23:23:14]: $ bundle init
[23:23:14]: $ echo 'gem "fastlane"' >> Gemfile
[23:23:14]: $ bundle install
[23:23:14]: 
[23:23:14]: After creating the Gemfile and Gemfile.lock, commit those files into version control
[23:23:14]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[23:23:17]: Using device frames version 'latest'
[23:23:17]: Framing screenshot './screenshots/fr-FR/iPhone6-01_RemoteTab-589996837d813aace56d9a0cf7b764a0.png'
[23:23:25]: Added frame: '/Users/Martijn/iMKapps/debug/Frameit/screenshots/fr-FR/iPhone6-01_RemoteTab-589996837d813aace56d9a0cf7b764a0_framed.png'
[23:23:25]: Framing screenshot './screenshots/background.png'
[23:23:26]: Could not get title for screenshot ./screenshots/background.png. Please provide one in your Framefile.json

**After**
Martijns-MacBook-Pro:Frameit Martijn$ fastlane frameit silver
[23:26:02]: Seems like launching fastlane takes a while - please run
[23:26:02]: 
[23:26:02]: $ [sudo] gem cleanup
[23:26:02]: 
[23:26:02]: to uninstall outdated gems and make fastlane launch faster
[23:26:02]: Alternatively it's recommended to start using a Gemfile to lock your dependencies
[23:26:02]: To get started with a Gemfile, run
[23:26:02]: 
[23:26:02]: $ bundle init
[23:26:02]: $ echo 'gem "fastlane"' >> Gemfile
[23:26:02]: $ bundle install
[23:26:02]: 
[23:26:02]: After creating the Gemfile and Gemfile.lock, commit those files into version control
[23:26:02]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[23:26:04]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables
[23:26:05]: Using device frames version 'latest'
[23:26:05]: Framing screenshot './screenshots/fr-FR/iPhone6-01_RemoteTab-589996837d813aace56d9a0cf7b764a0.png'
[23:26:13]: Added frame: '/Users/Martijn/iMKapps/debug/Frameit/screenshots/fr-FR/iPhone6-01_RemoteTab-589996837d813aace56d9a0cf7b764a0_framed.png'
[23:26:13]: Framing screenshot './screenshots/background.png'
[23:26:14]: Could not get title for screenshot ./screenshots/background.png. Please provide one in your Framefile.json
[23:26:14]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables
Martijns-MacBook-Pro:Frameit Martijn$ [23:26:14]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables
[23:26:14]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables

At the last it hang and I have to press CTRL-C to abort.

**Comment by milch** [comment]
@funnel20 Have you tried going to the link in the warning (https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables) and setting the correct variables? Does it still show this error?

**Comment by funnel20** [comment]
@milch I'm sorry, I didn't. I did now, but I have no clue how to get these 2 EXPORT line in what.
Forgive me, but I really liked the concept of `fastlane snapshot`, `fastlane frameit silver` and `fastlane deliver`. That's all I know of your fantastic tool chain. And of Terminal basically...

I'm on a MacBook Pro with macOS 10.13.1, so if you can guide me (or extend the referred documentation) that would be much appreciated.

**Comment by milch** [comment]
Open your terminal, and run the following code (copy exactly as shown here): 

```bash
echo -e "export LC_ALL=en_US.UTF-8\nexport LANG=en_US.UTF-8" >> ~/.bashrc
```

This will make sure all the necessary variables are set. Afterwards, restart your terminal, and the warnings should not be shown anymore

**Comment by funnel20** [comment]
@milch Thanks a lot, you're the man!
Problem solved.

I think it can help other users to add this to the docu. Or I might be dumbest...

**Comment by funnel20** [comment]
@milch Hmm, unfortunately the warning came back after several runs of frameit.
Any idea? I even restarted the MacBook.

**Comment by milch** [comment]
@funnel20 Do you still have this problem? Can you run `echo $LANG` from the terminal and see if the output is `en_US.UTF-8`? 

I would add the script I posted to the docs, but it does not work generally for all setups. I made some guesses as to what your setup might be and it seems to have worked, at least at first.

**Comment by funnel20** [comment]
@milch Thanks for getting back to me. It's still there.
The output of that command is nothing:
```
Martijns-MacBook-Pro:Frameit Martijn$ echo $LANG

Martijns-MacBook-Pro:Frameit Martijn$ 

```

**Comment by milch** [comment]
That is weird, can you also run `ls ~/.bashrc ~/.bash_profile ~/.zshrc ~/.config/fish/config.fish` and `echo $SHELL` and post the output please?

**Comment by funnel20** [comment]
Here you go:
```
Martijns-MacBook-Pro:Frameit Martijn$ ls ~/.bashrc ~/.bash_profile ~/.zshrc ~/.config/fish/config.fish
ls: /Users/Martijn/.config/fish/config.fish: No such file or directory
/Users/Martijn/.bash_profile	/Users/Martijn/.bashrc		/Users/Martijn/.zshrc
Martijns-MacBook-Pro:Frameit Martijn$ echo $SHELL
/bin/bash
Martijns-MacBook-Pro:Frameit Martijn$ 
```

**Comment by milch** [comment]
Hmm, that's interesting... Can you try running the following two scripts and then restarting your Terminal, to see if that fixes the locale settings for you? 

```
echo -e "export LC_ALL=en_US.UTF-8\nexport LANG=en_US.UTF-8" >> ~/.bash_profile
echo -e "export LC_ALL=en_US.UTF-8\nexport LANG=en_US.UTF-8" >> ~/.zshrc
```

**Comment by funnel20** [comment]
```
Martijns-MacBook-Pro:Frameit Martijn$ echo -e "export LC_ALL=en_US.UTF-8\nexport LANG=en_US.UTF-8" >> ~/.bash_profile
Martijns-MacBook-Pro:Frameit Martijn$ echo -e "export LC_ALL=en_US.UTF-8\nexport LANG=en_US.UTF-8" >> ~/.zshrc
Martijns-MacBook-Pro:Frameit Martijn$ echo $LANG

Martijns-MacBook-Pro:Frameit Martijn$ 
```

After restarting Terminal:
```
Martijns-MacBook-Pro:Frameit Martijn$ echo $LANG
en_US.UTF-8
Martijns-MacBook-Pro:Frameit Martijn$ 
```

And no more errors when running Frameit:
```
Martijns-MacBook-Pro:Frameit Martijn$ bundle exec fastlane frameit silver
[14:59:03]: Using device frames version 'latest'
[14:59:03]: Framing screenshot './screenshots/fr-FR/iPhone6-01_RemoteTab.png'
[14:59:10]: Added frame: '/Users/Martijn/iMKapps/debug/Frameit/screenshots/fr-FR/iPhone6-01_RemoteTab_framed.png'
[14:59:10]: Framing screenshot './screenshots/background.png'
[14:59:11]: Could not get title for screenshot ./screenshots/background.png. Please provide one in your Framefile.json
Martijns-MacBook-Pro:Frameit Martijn$ 
```
👍

**Comment by funnel20** [comment]
I hope this will now last longer then several days ago.

**Comment by milch** [comment]
Perfect! I don't know why, but it seems like your configuration file switched from `~/.bashrc` to `~/.bash_profile` within the last couple days. Maybe you installed an update, or some other tool that changed this? Honestly I've never seen a Mac using the `~/.bash_profile` file before, only linux, so I don't know what could've caused this

**Comment by funnel20** [comment]
The only change that I can think of is the High Sierra root bug that auto-installed on last Friday:
https://support.apple.com/en-us/HT208315

**Pull request merged**